src/components/ChallengeTile.tsx

```diff
@@ -0,0 +1,53 @@
+import {
+  Flex,
+  Button,
+  Text,
+  Stack,
+  Card,
+  CardBody,
+  CardFooter,
+  Image,
+  Heading,
+} from '@chakra-ui/react'
+import { NavLink } from '@/components/NavLink'
+import ChallengeProps from '@/types/ChallengeProps'
+
+export const ChallengeTile = ({
+  name,
+  image,
+  description,
+  difficulty,
+  slug,
+}: ChallengeProps) => {
+  return (
+    <Card maxW='sm' h='100%' _hover={{ boxShadow: '4px 4px 7px 0px #718096' }}>
+      <CardBody>
+        <Image src={image} alt={name} borderRadius='lg' />
+        <Stack mt='6' spacing='3'>
+          <Flex alignItems='center' justifyContent='space-between'>
+            <Heading size='md'>{name}</Heading>
+            <Text
+              color={
+                difficulty === 1
```

Comment:
Small recommendation here for easier readability and reuse :) Since enums automatically assign themselves numeric values, it might be a good option here.

```ts
enum DifficultyLevel {
  EASY,
  MEDIUM,
  HARD
}

console.log(DifficultyLevel.HARD) // 2
```

Also, for the colors, we could leverage a map to really simplify things, although this is a bit of overkill lol
```ts
const colorForDifficulty: Map<DifficultyLevel, string> = new Map([
  [DifficultyLevel.EASY]: 'green.600',
  ...
])

// In the JSX
colorForDifficulty(difficulty)
```

Reply:
I like the idea of using enums but I'm not sure I understand how to implement it.
1) the numeric values are coming from the database. Should I change 1,2,3 to EASY, MEDIUM and HARD on the Supabase table and then change the type for ChallengeProps to the enum you wrote above? On Supabase side I can create a new type as ENUM and change the 'difficulty' column type from int2 to the new one.
https://www.postgresql.org/docs/current/datatype-enum.html
2) as far as I know, Map objects can only be accessed with a .get() method, so it might be easier to just create a regular object and access it like so: colorForDifficulty[difficulty]. In that case, the key of the object can be a string or a number. I agree that this is a cleaner line of code than a nested ternary statement.

Reply:
> 1 

Hmmm that's a good question. On one hand an enum provides a better description of the difficulty level but on the other hand we could leverage numbers in the future to make something like a loading bar style display of difficulty level. I lean towards keeping the `int2` in supabase and using the `enum` to represent the numeric values in the frontend code. Something like this sandbox I think would work well :)

[![Edit DifficultyLevel](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/difficultylevel-g8t5vd?fontsize=14&hidenavigation=1&theme=dark)

> 2

I agree object notation is easier, maps are good for performance but in this situation it's just unnecessary haha. I included an example of the object with enum keys up in the codesandbox :)

Reply:
Ok, I understand now! Thanks for creating the codesandbox. I'll keep the table types as they are and add the enum to the front end.

Reply:
Added enums. I made them lower case because I think it looks better on the page. Uppercase looks good if the font is smaller. Let me know what you guys think!

Reply:
Did anyone else get kicked out of the MIT xPro Slack channel? I don't even see it on the list of my channels like it ceased to exist altogether... Should we create our own Slack channel or Discord?

Reply:
I was kicked too! I created a discord over in https://github.com/mitxpro-dev/learnjs/issues/47

Sorry I've been away for so long, back now to continue on the project!